inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/hook/MethodHookGenerator.java

```diff
@@ -89,52 +87,51 @@ private void addMetricsRecorder(MethodHookConfiguration config, CopyOnWriteArray
+    private void addDataProviderCalls(MethodReflectionInformation method, MethodHookConfiguration config, List<IHookAction> entryActions, List<IHookAction> exitActions) {
         config.getEntryProviders().forEach(pair -> {
             String dataKey = pair.getLeft();
             val providerCallConfig = pair.getRight();
             try {
-                BoundDataProvider call = generateAndBindDataProvider(hookedClass, dataKey, providerCallConfig);
+                BoundDataProvider call = generateAndBindDataProvider(method, dataKey, providerCallConfig);
                 entryActions.add(call);
             } catch (Exception e) {
-                log.error("Failed to build entry data provider {} for data {} on method {} of {}, no value will be assigned",
-                        providerCallConfig.getProvider().getName(), dataKey, signature, hookedClass.getName(), e);
+                log.error("Failed to build entry data provider {} for data {} on method {}.{}, no value will be assigned",
+                        providerCallConfig.getProvider().getName(), dataKey, method.getDeclaringClass().getName(), method.getName(), e);
             }
         });
 
         config.getExitProviders().forEach(pair -> {
             String dataKey = pair.getLeft();
             val providerCallConfig = pair.getRight();
             try {
-                BoundDataProvider call = generateAndBindDataProvider(hookedClass, dataKey, providerCallConfig);
+                BoundDataProvider call = generateAndBindDataProvider(method, dataKey, providerCallConfig);
                 exitActions.add(call);
             } catch (Exception e) {
-                log.error("Failed to build exit data provider {} for data {} on method {} of {}, no value will be assigned",
-                        providerCallConfig.getProvider().getName(), dataKey, signature, hookedClass.getName(), e);
+                log.error("Failed to build exit data provider {} for data {} on method {}.{}, no value will be assigned",
+                        providerCallConfig.getProvider().getName(), dataKey, method.getDeclaringClass().getName(), method.getName(), e);
             }
         });
     }
 
     /**
      * Generates a data provider and binds its arguments.
      *
-     * @param contextClass       the class in which this data provider will be used.
+     * @param method             the method in which this data provider will be used.
      * @param dataKey            the name of the data whose value is defined by executing the given data provider
      * @param providerCallConfig the specification of the call to the data provider
      * @return the executable data provider
      */
-    private BoundDataProvider generateAndBindDataProvider(Class<?> contextClass, String dataKey, DataProviderCallConfig providerCallConfig) {
+    private BoundDataProvider generateAndBindDataProvider(MethodReflectionInformation method, String dataKey, DataProviderCallConfig providerCallConfig) {
```

Comment:
Can you please name `method` to `methodInformation` in the methods of this class?